### PR TITLE
Add links between Deliver and admin panel

### PR DESCRIPTION
**Pull request by samuelhwilliams**
## 🎫 Ticket
BAU

## 📝 Description
We've had to manually URL hack to `/deliver/admin` to get to the platform admin panel so far; this patch adds links to the header to get back and forth between Deliver grant funding and the platform admin panel.

Also updates the admin panel header to show MHCLG Funding Service instead of GOV.UK

## 📸 Show the thing (screenshots, gifs)
image
image

### Review Readiness
- [x] PR title and description provide sufficient context for reviewers
- [x] Commits are logical, self-contained, and have clear descriptions

### Testing
- [x] I have tested this change and it meets the acceptance criteria for the ticket
- [x] I need the reviewer(s) to pull and run this change locally
- [-] New (non-developer) functionality has appropriate unit and integration tests
- [-] End-to-end tests have been updated (if applicable)
- [-] Edge cases and error conditions are tested

### CODE REVIEW

File: app/common/templates/common/partials/mhclg-header.html [hunk 1]
```diff
@@ -4,7 +4,16 @@
...
-  {%- set _navigation = params.navigation if params.navigation else [{ "text": ("Sign out"), "href": url_for('auth.sign_out') }] if currentUser and currentUser.is_authenticated else [] -%}
+  {%
+    set _navigation = (
+    params.navigation
+    if params.navigation else
+    [
+      {"text": "Admin panel", "href": url_for('platform_admin.index') } if authorisation_helper.is_platform_admin(currentUser) else {},
+      { "text": ("Sign out"), "href": url_for('auth.sign_out') }
+    ] if currentUser and currentUser.is_authenticated else []
+    )
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply app/common/templates/common/partials/mhclg-header.html]: doesn't cause any visual problems, am fine with it

**Review by srh-sloan — APPROVED** [review]
Great to have this in, nice little quality of life job!
